### ISSUE TITLE
Service Not Found - Help to debug

### ISSUE THREAD

**Issue by gepd** [jstasiak/python-zeroconf#85]
For a while I was using the [How do I se it](https://github.com/jstasiak/python-zeroconf#how-do-i-use-it) example to discover IoT devices as esp8266 (with `_arduino` as service name), but I'm having a problem since last week. None device is found when I run the example.

The first I thought, it was problem of my device, but when I run the Arduino IDE, it's listed as mDNS service.

In the Arduino's github shows that `_arduino` is the name of the service: https://github.com/esp8266/Arduino/blob/master/libraries/ESP8266mDNS/ESP8266mDNS.cpp#L891

I've disabled my firewall thinking it may be the problem, but same results. And now other people using nodemcu devices have  reported me the same issue, so I guess isn't my configuration. 

My question is, do you have any recomendation to know where is the problem?

I've used `--debug` as argument but the output isn't very clear to me:

`DEBUG:zeroconf:Sending <zeroconf.DNSOutgoing object at 0x02A1A690> as '\x00\x00\x00\x00\x00\x01\x00\x00\x00\x00\x00\x00\x08_arduino\x04_tcp\x05local\x00\x00\x0c\x00\x01'...`

**Comment by stephenrauch** [comment]
Wireshark

Use a filter of ip.addr == 224.0.0.251

**Comment by gepd** [comment]
It's showing this:

```
Frame 12785: 333 bytes on wire (2664 bits), 333 bytes captured (2664 bits) on interface 0
    Interface id: 0 (\Device\NPF_{CBF5CCC9-6101-487D-9AD1-ED44CD56F67A})
    Encapsulation type: Ethernet (1)
    Arrival Time: Oct  1, 2016 22:53:48.119158000
    [Time shift for this packet: 0.000000000 seconds]
    Epoch Time: 1475373228.119158000 seconds
    [Time delta from previous captured frame: 0.081998000 seconds]
    [Time delta from previous displayed frame: 0.081998000 seconds]
    [Time since reference or first frame: 500.777982000 seconds]
    Frame Number: 12785
    Frame Length: 333 bytes (2664 bits)
    Capture Length: 333 bytes (2664 bits)
    [Frame is marked: False]
    [Frame is ignored: False]
    [Protocols in frame: eth:ethertype:ip:udp:mdns]
    [Coloring Rule Name: UDP]
    [Coloring Rule String: udp]
Ethernet II, Src: Espressi_d3:f8:a9 (18:fe:34:d3:f8:a9), Dst: IntelCor_ee:73:02 
    Destination: IntelCor_ee:73:02
        Address: IntelCor_ee:73:02
        .... ..0. .... .... .... .... = LG bit: Globally unique address (factory default)
        .... ...0 .... .... .... .... = IG bit: Individual address (unicast)
    Source: Espressi_d3:f8:a9 (18:fe:34:d3:f8:a9)
        Address: Espressi_d3:f8:a9 (18:fe:34:d3:f8:a9)
        .... ..0. .... .... .... .... = LG bit: Globally unique address (factory default)
        .... ...0 .... .... .... .... = IG bit: Individual address (unicast)
    Type: IPv4 (0x0800)
Internet Protocol Version 4, Src: 192.168.1.108, Dst: 224.0.0.251
    0100 .... = Version: 4
    .... 0101 = Header Length: 20 bytes (5)
    Differentiated Services Field: 0x00 (DSCP: CS0, ECN: Not-ECT)
        0000 00.. = Differentiated Services Codepoint: Default (0)
        .... ..00 = Explicit Congestion Notification: Not ECN-Capable Transport (0)
    Total Length: 319
    Identification: 0x0020 (32)
    Flags: 0x00
        0... .... = Reserved bit: Not set
        .0.. .... = Don't fragment: Not set
        ..0. .... = More fragments: Not set
    Fragment offset: 0
    Time to live: 1
        [Expert Info (Note/Sequence): "Time To Live" != 255 for a packet sent to the Local Network Control Block (see RFC 3171)]
            ["Time To Live" != 255 for a packet sent to the Local Network Control Block (see RFC 3171)]
            [Severity level: Note]
            [Group: Sequence]
    Protocol: UDP (17)
    Header checksum: 0x157f [validation disabled]
    [Header checksum status: Unverified]
    Source: 192.168.1.108
    Destination: 224.0.0.251
    [Source GeoIP: Unknown]
    [Destination GeoIP: Unknown]
User Datagram Protocol, Src Port: 5353, Dst Port: 5353
    Source Port: 5353
    Destination Port: 5353
    Length: 299
    Checksum: 0x632f [unverified]
    [Checksum Status: Unverified]
    [Stream index: 45]
Multicast Domain Name System (response)
    Transaction ID: 0x0000
    Flags: 0x8400 Standard query response, No error
        1... .... .... .... = Response: Message is a response
        .000 0... .... .... = Opcode: Standard query (0)
        .... .1.. .... .... = Authoritative: Server is an authority for domain
        .... ..0. .... .... = Truncated: Message is not truncated
        .... ...0 .... .... = Recursion desired: Don't do query recursively
        .... .... 0... .... = Recursion available: Server can't do recursive queries
        .... .... .0.. .... = Z: reserved (0)
        .... .... ..0. .... = Answer authenticated: Answer/authority portion was not authenticated by the server
        .... .... ...0 .... = Non-authenticated data: Unacceptable
        .... .... .... 0000 = Reply code: No error (0)
    Questions: 0
    Answer RRs: 4
    Authority RRs: 0
    Additional RRs: 0
    Answers
        _arduino._tcp.local: type PTR, class IN, esp8266-d3f8a9._arduino._tcp.local
            Name: _arduino._tcp.local
            Type: PTR (domain name PoinTeR) (12)
            .000 0000 0000 0001 = Class: IN (0x0001)
            0... .... .... .... = Cache flush: False
            Time to live: 4500
            Data length: 36
            Domain Name: esp8266-d3f8a9._arduino._tcp.local
        esp8266-d3f8a9._arduino._tcp.local: type TXT, class IN
            Name: esp8266-d3f8a9._arduino._tcp.local
            Type: TXT (Text strings) (16)
            .000 0000 0000 0001 = Class: IN (0x0001)
            0... .... .... .... = Cache flush: False
            Time to live: 4500
            Data length: 56
            TXT Length: 12
            TXT: tcp_check=no
            TXT Length: 13
            TXT: ssh_upload=no
            TXT Length: 13
            TXT: board=generic
            TXT Length: 14
            TXT: auth_upload=no
        esp8266-d3f8a9._arduino._tcp.local: type SRV, class IN, cache flush, priority 0, weight 0, port 8266, target esp8266-d3f8a9.local
            Service: esp8266-d3f8a9
            Protocol: _arduino
            Name: _tcp.local
            Type: SRV (Server Selection) (33)
            .000 0000 0000 0001 = Class: IN (0x0001)
            1... .... .... .... = Cache flush: True
            Time to live: 120
            Data length: 28
            Priority: 0
            Weight: 0
            Port: 8266
            Target: esp8266-d3f8a9.local
        esp8266-d3f8a9.local: type A, class IN, cache flush, addr 192.168.1.108
            Name: esp8266-d3f8a9.local
            Type: A (Host Address) (1)
            .000 0000 0000 0001 = Class: IN (0x0001)
            1... .... .... .... = Cache flush: True
            Time to live: 120
            Data length: 4
            Address: 192.168.1.108

```

**Comment by gepd** [comment]
I've tried in a Linux environment and same results.

The log of wireshark seems very normal to me, could it be a problem of zeroconf?

**Comment by jackha** [comment]
I got a similar problem with the ServiceBrowser not finding any devices. The cause was one of zeroconf's dependencies: netifaces. Installing the latest version will download version 0.10.5, but with this version it breaks my ServiceBrowser by not finding anything. Downgrading to 0.10.4 and everything works fine.

pip uninstall netifaces
pip install netifaces==0.10.4 

ps. I got a windows 32 bit environment, but I think the problem may be platform independent.
